src/App.tsx

```diff
@@ -1,27 +1,57 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getAll, getFirstFive, getAllRed } from './api/goods';
+import { Good } from './types/Good';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+  function getAllGoods() {
+    getAll()
+      .then(resultAll => setGoods(resultAll));
+  }
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+  function getFive() {
+    getFirstFive()
+      .then(resultFive => setGoods(resultFive));
+  }
 
-    <button type="button" data-cy="red-button">
-      Load red goods
-    </button>
+  function getRed() {
+    getAllRed()
+      .then(resultRed => setGoods(resultRed));
+  }
```

Comment:
you are duplicating the logic several times, it is better to put it in a separate function and pass different arrays to its parameter when you call it for different buttons